src/main/java/lotto/domain/LottoGame.java

```diff
@@ -0,0 +1,63 @@
+package lotto.domain;
+
+import java.util.*;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toList;
+
+public class LottoGame {
+    public static final int LOTTO_PRICE = 1000;
+    public static final int LOTTO_NUMBER_BOUNDARY = 45;
+    public static final int MINIMUM_LOTTO_NUMBER = 1;
+    public static final int LOTTO_NUMBER_SIZE = 6;
+
+    public int match(Lotto lotto, Lotto winningLotto) {
+        return (int) lotto.getNumbers().stream()
+                .filter(num -> winningLotto.getNumbers().contains(num))
+                .count();
+    }
+
+    public Rank getRank(int matchCount) {
+        return Rank.of(matchCount);
+    }
+
+    public int purchaseLotto(int price) {
+        if (price < LOTTO_PRICE) {
+            throw new IllegalArgumentException("구매 금액이 로또 한 장 값보다 작을 수 없습니다.");
+        }
+        return price / LOTTO_PRICE;
+    }
+
+    public int getTotalWinnings(List<Rank> winningResults) {
+        return winningResults.stream()
+                .mapToInt(Rank::getWinningMoney)
+                .sum();
+    }
+
+    public double getRateOfRevenue(int price, int totalWinnings) {
+        return (double) totalWinnings / price;
+    }
+
+    public List<Lotto> createLottos(int purchaseCount) {
+        return Stream
+                .generate(() -> new Lotto(generateNonDuplicateNumbers()))
```

Comment:
```suggestion
                .generate(() -> new Lotto(numbers.of()))
```

```java
public interface Numbers {
    List<Integer> of();
}

public class LottoNumbers implements Numbers {
    public static final int LOTTO_NUMBER_BOUNDARY = 45;
    public static final int MINIMUM_LOTTO_NUMBER = 1;
    public static final int LOTTO_NUMBER_SIZE = 6;
    
    Random random = new Random();

    @Override
    public List<Integer> of() {
        List<Integer> numbers = generate();
        shuffle(numbers);
        return numbers;
    }

    private List<Integer> generate() {
        Set<Integer> numbers = new HashSet<>();
        while (numbers.size() != LOTTO_NUMBER_SIZE) {
            numbers.add(generateRandomNumber());
        }
        return numbers.stream()
                .collect(toList());
    }

    private int generateRandomNumber() {
        return random.nextInt(LOTTO_NUMBER_BOUNDARY) + MINIMUM_LOTTO_NUMBER;
    }
}
```

LottoGame에서 숫자 생성 역할도 가지기보다는, 의존성을 외부에서 주입하는 것은 어떨까요